build_tools/benchmarks/reporting/parse_shark_benchmarks.py

```diff
@@ -0,0 +1,308 @@
+      _IREE_LATENCY, _SHARK_LATENCY, _IREE_VS_BASELINE, _SHARK_VS_BASELINE,
+      _IREE_VS_SHARK
+  ])
+
+  models = df_iree.model.unique()
+  for model in models:
+    iree_results_per_model = df_iree.loc[df_iree.model == model]
+    dialects = iree_results_per_model.dialect.unique()
+    for dialect in dialects:
+      iree_results_per_dialect = iree_results_per_model.loc[
+          iree_results_per_model.dialect == dialect]
+      data_types = iree_results_per_dialect.data_type.unique()
+      for data_type in data_types:
+        iree_results_per_datatype = iree_results_per_dialect.loc[
+            iree_results_per_dialect.data_type == data_type]
+        device_types = iree_results_per_datatype.device.unique()
+        for device in device_types:
+          iree_results = iree_results_per_datatype.loc[
+              iree_results_per_datatype.device == device]
+          if len(iree_results) != 3:
+            print("Warning! Expected number of results to be 3. Got " +
+                  str(len(iree_results)))
+            print(str(iree_results))
+            continue
+
+          baseline_results = df_baseline.loc[(df_baseline.model == model) &
+                                             (df_baseline.dialect == dialect) &
+                                             (df_baseline.data_type
+                                              == data_type) &
+                                             (df_baseline.device == device)]
+
+          if baseline_results.empty:
+            # We use snapshots of latencies for baseline. If it is a new
+            # benchmark that is not included in the snapshot yet, emit a
+            # warning.
+            print(
+                "Warning: No baseline results found for " + model + ", " +
+                dialect + ", " + data_type + ", " + device +
+                ". Using IREE version as baseline. Please update baseline csv.")
+            engine = iree_results.engine.iloc[0]
+            baseline_latency = iree_results.loc[iree_results.engine == engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+          else:
+            engine = baseline_results.engine.iloc[0]
+            baseline_latency = baseline_results.loc[baseline_results.engine ==
+                                                    engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+
+          iree_latency = iree_results.loc[iree_results.engine == "shark_iree_c"]
+          iree_latency = iree_latency.iloc[0]["ms/iter"]
+          iree_vs_baseline = html_utils.format_latency_comparison(
+              iree_latency, baseline_latency)
+
+          if df_shark is not None:
+            shark_results = df_shark.loc[(df_shark.model == model) &
+                                         (df_shark.dialect == dialect) &
+                                         (df_shark.data_type == data_type) &
+                                         (df_shark.device == device)]
```

Comment:
I think this is doing a bitwise and of the booleans after presumably converting them to ints, which does give the same result, but is a bit weird, since I think you really just want the logical and of booleans (Pandas is looking up a boolean array with `loc`, right?)

```suggestion
            shark_results = df_shark.loc[(df_shark.model == model) and
                                         (df_shark.dialect == dialect) and
                                         (df_shark.data_type == data_type) and
                                         (df_shark.device == device)]
```

Reply:
Using `and` gives an error. Some background: https://stackoverflow.com/questions/21415661/logical-operators-for-boolean-indexing-in-pandas